### ISSUE TITLE
GameGuru MAX - Request for the Future - "Decals" with "Depth"

### ISSUE THREAD

**Issue by MonkeyFrogStudio** [TheGameCreators/GameGuruRepo#4620]
I apologize, I don't recall the correct words for what I am requesting here, but other game engines have decals that have "depth" to them. Instead of a polygonal plane that the decal sits on and is rotated into position, a decal is PROJECTED a specific depth and effects everything that is within that projected area. This is often represented by a 3D wireframe cube to show how big and how "deep" the decal projection goes. The reason this is great is because the "decal" is no longer flat, but can be projected onto surfaces, even around corners, and even on both a wall and a floor at the same time. 

Here is a video of a quite advanced system similar to this used in the remake of Deadspace:

https://www.monkeyfrogstudio.com/uploads/deadspace_shader_decal.mp4

Here's an example from the Unreal documentation on Deferred Decals:

https://docs.unrealengine.com/4.27/en-US/Resources/ContentExamples/Decals/1_1/

And in Unity:

https://docs.unity3d.com/Packages/com.unity.render-pipelines.universal@12.0/manual/images/decal/decal-projector-scene-view.png

https://docs.unity3d.com/Packages/com.unity.render-pipelines.universal@12.0/manual/renderer-feature-decal.html

I don't know if the Wicked Engine supports these types of decals but if it does, these sorts of decals would be a very nice addition to MAX in the future.

**Comment by bond1mjb** [comment]
@MonkeyFrogStudio Yes indeed Wicked has this, and yes it's very cool.  The decal can wrap around curved surfaces.  I think sometimes this is referred to as "splatting" decals?

Here I'm playing around with it in the Wicked Editor:

https://user-images.githubusercontent.com/87768878/236999489-333cfe33-a95b-49cc-b6fb-7af3699027a7.mp4

**Comment by bond1mjb** [comment]
@MonkeyFrogStudio  And another vid where it's being projected onto both the wall and floor:

https://user-images.githubusercontent.com/87768878/237000398-93215914-f676-4997-a893-89b517ac7c5f.mp4

**Comment by MonkeyFrogStudio** [comment]
@bond1mjb Thanks for sharing these! Since this is already in Wicked, then it should be pretty simple to implement this in MAX. It would be a lot better than the current "decals" that MAX is using ... certainly a lot more flexible. And if, like in the Deadspace example, we can combine this with shaders, it would open up an entire world of possibilities.

**Comment by Granada1** [comment]
Also help with bullet strikes in corners and on edges 👍

**Comment by mav3r1ck198-1** [comment]
this would be awesome, roll on the wicked update please!

**Comment by bond1mjb** [comment]
Yes it's the default way Wicked uses decals, no complex user setup required, so I would think it would be easy to implement into MAX.  But I really don't know for sure.

You can also assign the entire PBR texture set to them as well just like any other assets.  So like in the Dead Space example you could use them for added dirt, glistening frost, damage from bullet hits, blood splatters, etc.

**Comment by fearlesswee** [comment]
It appears I was well ahead of the curve with suggesting this lol!

#711

**Comment by MonkeyFrogStudio** [comment]
I wanted to add to this request. Decal projection, in and of itself, would be great to have. But many engines go a bit further with it by allowing the decals to be full PBR materials. This means that the decal could have, not only color information, but also normal, metallic, roughness, and AO information as well. This is important for a wide variety of reasons. For example, a decal doesn't have to be just a texture image on a wall or floor, like some kind of wall poster. A decal could be damage, such as cracks or chunks missing due to age, etc., with 'depth' via the normal map; deeper exposed surfaces (like plaster removed from a wall) via color, normal, etc. This means that the end-user could use decals to add things like broken plaster, cracks, etc., to tiling wall textures to further break up the pattern and add details.

![damage](https://github.com/TheGameCreators/GameGuruRepo/assets/36735507/be3387fd-4b50-422a-9714-bfdeeac5f125)

![damage2](https://github.com/TheGameCreators/GameGuruRepo/assets/36735507/c267e0af-651a-418b-8560-bcea696d229f)

![damage3](https://github.com/TheGameCreators/GameGuruRepo/assets/36735507/d0d08205-7a2b-4b5d-90a0-d1da2208e0b8)

Not only that, but things like blood splatter would react to light correctly due to the metallic and roughness settings and, due to a normal map, not just be perfectly flat against the object upon which they are projected.

**Comment by synchromesh62** [comment]
Am i misreading this ?
I was creating PBR textures for decals long ago and they react great ?

https://github.com/TheGameCreators/GameGuruRepo/assets/35708217/afe6c121-085a-4a3b-b225-b8b48b7258e8

**Comment by MonkeyFrogStudio** [comment]
@synchromesh62 No, you're not misreading this, but you need to read it in CONTEXT. The context here is DECAL PROJECTION, so that decals conform to all surfaces and not just flat ones.

**Comment by OldFlak** [comment]
Yeah, just do it already :)
